connect/src/protocols/tokenTransfer.ts

```diff
@@ -447,57 +455,59 @@ export class TokenTransfer<N extends Network = Network>
       ? await srcChain.getNativeWrappedTokenId()
       : transfer.token;
 
-    const dstDecimals = await dstChain.getDecimals(dstToken.address);
     const srcDecimals = await srcChain.getDecimals(srcToken.address);
+    const dstDecimals = await dstChain.getDecimals(dstToken.address);
+
+    const srcAmount = amountFromBaseUnits(transfer.amount, srcDecimals);
 
-    // truncate the amount for over-the-wire max representation
-    const truncate = (amt: bigint, decimals: bigint) =>
-      decimals > 8 ? amt / 10n ** (decimals - 8n) : amt;
-    // scale by number of decimals, pair with truncate
-    const scale = (amt: bigint, decimals: bigint) =>
-      decimals > 8 ? amt * 10n ** (decimals - 8n) : amt;
-    // dedust for source using the number of decimals on the source chain
-    // but scale back to original decimals
-    const dedust = (amt: bigint) => scale(truncate(amt, srcDecimals), srcDecimals);
-    // truncate and rescale to fit destination token
-    const mintable = (amt: bigint) => scale(truncate(amt, srcDecimals), dstDecimals);
-
-    // dedust for transfer over the bridge which truncates to 8 decimals
-    let srcAmount = dedust(transfer.amount);
-    let dstAmount = mintable(transfer.amount);
+    const srcAmountTruncated = truncateAmount(srcAmount, TOKEN_BRIDGE_MAX_DECIMALS);
+    const dstAmountTruncated = scaleAmount(srcAmountTruncated, dstDecimals);
```

Comment:
this is correct for math but nitpicking the var name, maybe call it `dstAmountMintable` or something?